src/backend/distributed/planner/distributed_planner.c

```diff
@@ -556,6 +557,9 @@ CreateDistributedPlannedStmt(uint64 planId, PlannedStmt *localPlan, Query *origi
 		RaiseDeferredError(distributedPlan->planningError, ERROR);
 	}
 
+	/* traverse distributedPlan and extract the list of used subplans */
+	FindSubPlansUsedInPlan(distributedPlan);
```

Comment:
This API is confusing, maybe we can do `distributedPlan->usedSubPlanNodeList = FindSubPlansUsedInPlan(distributedPlan);`?

Reply:
I suggested the same, but `distributedPlan->usedSubPlanNodeList = findSubPlansUsedInQuery(originalQuery)`, because that seems safer than looking into `jobQuery` or `insertSelectQuery` etc

Reply:
bump

Reply:
I am working on this one now. Will get it working before merging the PR

Reply:
Marco's suggestion is completed yesterday. I was not able to use `originalQuery` instead ob `distributedPlan->jobQuery` or `distributedPlan->insertSelectQuery` though.